src/pubsub-publisher/pubsub-publisher.service.ts

```diff
@@ -37,36 +37,36 @@ export class PubSubPublisherService {
     data: Buffer,
     attributes: Record<string, string>
   ): Promise<string> {
-    const timeout = setTimeoutAsync(this.settings.requestTimeoutMillis);
+    const publisherClient = new v1.PublisherClient({});
+    const formattedTopic = publisherClient.projectTopicPath(projectId, topic.name);
+    const dataBuffer = Buffer.from(data);
+    const messageElement = {
+      data: dataBuffer,
+    };
+    const messages = [messageElement];
+    const request = {
+      topic: formattedTopic,
+      messages,
+      attributes,
+    };
 
-    const [result] = await Promise.race([
-      topic.publishMessage({ data, attributes }),
-      timeout.then(() => {
-        throw new PubSubTimeoutError();
-      }),
-    ]);
+    const [{ messageIds }] = await publisherClient.publish(request, {
+      retry: {
+        backoffSettings: {
+          initialRetryDelayMillis: 1,
+          retryDelayMultiplier: 1.3,
+          maxRetryDelayMillis: 100,
+          totalTimeoutMillis: this.settings.requestTimeoutMillis,
+        },
+      },
+    });
 
-    timeout.clear();
+    const [id] = messageIds || [''];
 
-    return result;
+    return id;
```

Comment:
`return messageIds[0] ?? '';` would be better

Reply:
agreed